_posts/2024-05-01-develop-guideline-multiple-data-source-in-opensearch-and-plugins.md

```diff
@@ -0,0 +1,341 @@
+
+### 2.2 The data source as a saved object
+
+The data source is persisted as a saved object in the Dashboards system index. At the time of creation, a UUID is assigned as its unique identifier and data source metadata is also gathered, including the remote cluster version and associated plugins. This automatically generated data, as well as user-provided information including the title, endpoint, and authentication methods, are stored as attributes of the data source object. For the current schema of the object, refer to the [related code](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/data_source/common/data_sources/types.ts#L8). 
+
+## 3 FAQ 
+
+Q: Is any UI component provided by `dataSourceManagemet` used in core OpenSearch?
+A: A data source selector is now used in Dev Tools, the tutorial, and on the saved object management page. You can see the component after setting `data_source.enabled` to true in `opensearch_dashbards.yml`.
+
+Q: Given that a dashboard is a saved object itself, will it not have a data connection associated with it?
+A: A data source, represented as a saved object, is identified by its unique saved object ID. Saved objects created from data sources have a references section containing the ID, name, and type. This reference helps you to identify the associated data source for the saved object. Here is an example of a index pattern reference:
+```
+"references": [
+    {
+        "id": "9dc4e190-08a0-11ef-901b-bb0301313422",
+        "type": "data-source",
+        "name": "dataSource"
+    }
+]
+```
+The visualization reference is slightly different because it points to the related index pattern and uses the index pattern ID to find the corresponding data source: 
+```
+"references": [
+    {
+        "name": "kibanaSavedObjectMeta.searchSourceJSON.index",
+        "type": "index-pattern",
+        "id": "f1613e10-08a6-11ef-901b-bb0301313422"
+        }
+]
+```
+
+
+## 4 List of supported plugins/features
+
+Starting in version 2.14, multiple data sources are supported in nine external Dashboards plugins: 
+
+* Index Management
+* Machine Learning
+* Search Relevance
+* Anomaly Detection
+* Maps
+* Security
+* Notification
+* Query Workbench
+* Trace Analytics 
+
+Additionally, we've extended this functionality to two core visualizations: Time Series Visual Builder (TSVB) and Region Map.
+
+
+## 5 Conclusion
+
+In this blog post, we've covered the process of integrating OpenSearch Dashboards plugins with multiple data sources. We strongly encourage you to experiment with multiple data sources using the example plugins provided above. Thanks to Lu Yu for developing the example plugin! This exploration will enable you to experience the benefits of accessing multiple data sources within a single dashboard as well as allow your plugin to expose a wider range of data from remote clusters.
```

Comment:
```suggestion
This blog post covered the process of integrating OpenSearch Dashboards plugins with multiple data sources. We encourage you to experiment with multiple data sources using the provided example plugins, developed by Lu Yu. This will allow you to experience the benefits of accessing multiple data sources within a single dashboard and enable your plugin to expose a wider range of data from remote clusters.
```